Microsoft.Toolkit.Uwp.UI.Controls/RangeSelector/RangeSelector.cs

```diff
@@ -639,19 +665,192 @@ public double StepFrequency
+            {
+                return;
+            }
+
+            rangeSelector.DrawTicks();
+        }
+
+        /// <summary>
+        /// Gets or sets the value for tick placement position.
+        /// </summary>
+        /// <value>
+        /// The value for tick placement.
+        /// </value>
+        public TickPlacement TickPlacement
+        {
+            get
+            {
+                return (TickPlacement)GetValue(TickPlacementProperty);
+            }
+
+            set
+            {
+                SetValue(TickPlacementProperty, value);
+            }
+        }
+
+        private static void TickPlacementChangedCallback(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var rangeSelector = d as RangeSelector;
+            if (rangeSelector == null)
+            {
+                return;
+            }
+
+            rangeSelector.DrawTicks();
+        }
+
+        /// <summary>
+        /// Gets or sets the value for tick offset.
+        /// </summary>
+        /// <value>
+        /// The value for tick offset.
+        /// </value>
+        public double TickOffset
+        {
+            get
+            {
+                return (double)GetValue(TickOffsetProperty);
+            }
+
+            set
+            {
+                SetValue(TickOffsetProperty, value);
+            }
+        }
+
+        private static void TickOffsetChangedCallback(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var rangeSelector = d as RangeSelector;
+            if (rangeSelector == null)
+            {
+                return;
+            }
+
+            rangeSelector.DrawTicks();
+        }
+
+        /// <summary>
+        /// Gets or sets the value for snapping to steps or tick frequency.
+        /// </summary>
+        /// <value>
+        /// The value for snapping to ticks or steps.
+        /// </value>
+        public SliderSnapsTo SnapsTo
+        {
+            get
+            {
+                return (SliderSnapsTo)GetValue(SnapsToProperty);
+            }
+
+            set
+            {
+                SetValue(SnapsToProperty, value);
+            }
+        }
+
+        private void DrawTicks()
+        {
+            if (_tickCanvas == null)
+            {
+                return;
+            }
+
+            _tickCanvas.Children.Clear();
+
+            if (TickPlacement == TickPlacement.None)
+            {
+                return;
+            }
+
+            double tickMargin = 3;
+            double tickHeight = 4;
+            double tickActualStep = _containerCanvas.ActualWidth / (Maximum - Minimum);
+            double relativeTopForInlineTicks = (_containerCanvas.ActualHeight / 2) - (tickHeight / 2);
+            double relativeTopForTopTicks = (_containerCanvas.ActualHeight / 2) - (tickMargin + tickHeight);
+            double relativeTopForBottomTicks = (_containerCanvas.ActualHeight / 2) + tickMargin;
```

Comment:
why not do this calculation when needed in the appropriate block below. Only one will be needed at a time.

Reply:
I placed them in each appropriate block initially, then I thought they would be calculated many times, once for each tick, so I was thinking its better for performance, what do you think?

Reply:
Yup, you are right. To optimize even further, you already know the TickPlacement at this point, so you can only do the if statements up here